components/eamxx/src/physics/mam/eamxx_mam_optics_process_interface.cpp

```diff
@@ -1,110 +1,531 @@
+      map_table_name_species_id["bc"] = 5;  // bc :black-c
+      map_table_name_species_id["mom"] = 8;   // mom:m-organic
+
+      for (const auto& item : map_table_name_species_id) {
+       const auto spec_name = item.first;
+       const int species_id = item.second;
+       const auto table_name = "mam4_" + spec_name + "_physical_properties_file";
+       const auto& fname = m_params.get<std::string>(table_name);
+        // read data
+        // need to update table name
+        params_aero.set("Filename", fname);
+        AtmosphereInput refindex_aerosol(params_aero, grid_, host_views_aero,
+                                         layouts_aero);
+        refindex_aerosol.read_variables();
+        refindex_aerosol.finalize();
+        // copy data to device
+        mam_coupling::set_refindex_aerosol(
+            species_id, host_views_aero,
+            specrefndxsw_host,  // complex refractive index for water visible
+            specrefndxlw_host);
+      }  // done ispec
+      // reshape specrefndxsw_host and copy it to device
+      mam4::modal_aer_opt::set_device_specrefindex(
+          aerosol_optics_device_data_.specrefindex_sw, "short_wave",
+          specrefndxsw_host);
+      mam4::modal_aer_opt::set_device_specrefindex(
+          aerosol_optics_device_data_.specrefindex_lw, "long_wave",
+          specrefndxlw_host);
+    }
   }
+  //FIXME: We are hard-coding the band ordering in RRTMGP.
+  //TODO: We can update optics file using the ordering below (rrtmg_to_rrtmgp_swbands_).
+  // Mapping from old RRTMG sw bands to new band ordering in RRTMGP
+  // rrtmg_swband (old) = 2925, 3625, 4325, 4900, 5650, 6925, 7875, 10450, 14425, 19325,
+    // 25825, 33500, 44000, 1710 ;
+  // rrtmgp_swband (new) = 1710, 2925, 3625, 4325, 4900, 5650, 6925, 7875, 10450, 14425,
+    // 19325, 25825, 33500, 44000 ;
+    // given the rrtmg index return the rrtmgp index
+  std::vector<int> temporal = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 0};
+  auto get_idx_rrtmgp_from_rrtmg_swbands_host =mam_coupling::view_int_1d::HostMirror(temporal.data(),nswbands_);
+  get_idx_rrtmgp_from_rrtmg_swbands_ = mam_coupling::view_int_1d("rrtmg_to_rrtmgp_swbands",nswbands_);
+  Kokkos::deep_copy(get_idx_rrtmgp_from_rrtmg_swbands_,get_idx_rrtmgp_from_rrtmg_swbands_host);
 }
+void MAMOptics::run_impl(const double dt) {
+
+  constexpr Real zero=0.0;
+  constexpr Real one=1.0;
+
+  const auto policy =
+      ekat::ExeSpaceUtils<KT::ExeSpace>::get_default_team_policy(ncol_, nlev_);
+  const auto scan_policy = ekat::ExeSpaceUtils<
+      KT::ExeSpace>::get_thread_range_parallel_scan_team_policy(ncol_, nlev_);
+
+  // preprocess input -- needs a scan for the calculation of atm height
+  Kokkos::parallel_for("preprocess", scan_policy, preprocess_);
+  Kokkos::fence();
+
+  //tau_w_g : aerosol asymmetry parameter * tau * w
+  const auto aero_tau_g_sw = get_field_out("aero_tau_g_sw_mam4").get_view<Real ***>();
```

Comment:
If this is actually asymmetry parameter * tau * w, can we call it `aero_tau_ssa_g_sw` instead of `aero_tau_g_sw` for consistency?

Reply:
Here, I am using the Fortran code as a reference to name this variable;[ see here](https://github.com/eagles-project/e3sm_mam4_refactor/blob/f0ef4b66dcb8a3aaae1261e3d3ef1a6b320e277b/components/eam/src/physics/cam/aer_rad_props.F90#L100).  For consistency, I will remove `aero` from the name of this variable.

Reply:
From that section of code:
```
    real(r8),         intent(out) :: tau    (pcols,0:pver,nswbands) ! aerosol extinction optical depth
    real(r8),         intent(out) :: tau_w  (pcols,0:pver,nswbands) ! aerosol single scattering albedo * tau
    real(r8),         intent(out) :: tau_w_g(pcols,0:pver,nswbands) ! aerosol assymetry parameter * tau * w
    real(r8),         intent(out) :: tau_w_f(pcols,0:pver,nswbands) ! aerosol forward scattered fraction * tau * w
```
Note that the assymetry parameter one is `tau_w_g`, which indicates that it is tau * w * g. What you have is `tau_g`, which would suggest that it is just tau * g, not tau * w * g. This is confusing.